transport/kp/key_provider.go

```diff
@@ -241,9 +244,30 @@ func (sp *StandardProvider) Generate(algo string, size int) (err error) {
 		}
 		sp.internal.keyPEM = pem.EncodeToMemory(p)
 
+		sp.internal.priv = priv
+	case "ed25519":
+		if size != (ed25519.PublicKeySize * 8) {
```

Comment:
The size parameter doesn't refer to the actual size of the public key, does it?

Reply:
Oh, yes.. this should probably refer to the private key size. I'll change this.

Reply:
This is counterintuitively right. ed25519.PrivateKeySize accounts for both the public key (32 bytes) and the seed (32 bytes), totaling 64 bytes. ed25519.go

However, I do not see the point of specifying the key size, since ed25519 is always the same, as I know. I think the field size should be simply ignored when generating ed25519 keys.

Reply:
This is counterintuitively right. ed25519.PrivateKeySize accounts for both the public key (32 bytes) and the seed (32 bytes), totaling 64 bytes. [ed25519.go](https://github.com/golang/go/blob/8c0256b398f3b23f24a1a3ac0cb00c2d8fb7f506/src/crypto/ed25519/ed25519.go#L154-L170)

**However**, I do not see the point of specifying the key size, since ed25519 is always the same, as I know. I think the field **size** should be simply ignored when generating ed25519 keys.